### PR TITLE
Add open telemetry instrumentation to template files

### PR DESCRIPTION
**Pull request by HuwONS**
### What

Added open telemetry to the template generation functionality

### How to review

Check the templates generate as expected

### Who can review

Fran Moore, Chales Colbourn, Gordon Kerr

### Mentions
@HuwONS @gordonkerrONS @franmoore05 @hhsons @charlescolbourn

### CODE REVIEW

File: project_generation/content/templates/api/main.go.tmpl
Comment (charlescolbourn) [inline project_generation/content/templates/api/main.go.tmpl:70]: Fatal & return - probably should not be the case, we don't want services to break if they don't have otel config

File: project_generation/content/templates/event/service/service.go.tmpl
Comment (charlescolbourn) [inline project_generation/content/templates/event/service/service.go.tmpl:44]: check with @gordonkerrONS here. We want to have his kafka changes in, so maybe this needs to be blocked until that's merged & released.

File: project_generation/content/templates/event/service/service.go.tmpl
Comment (franmoore05) [inline project_generation/content/templates/event/service/service.go.tmpl:40]: When I generate a new event driven project, I get an error "undefined: router"

**Comment by franmoore05** [comment]
I think it would also be useful to instrument the controller generation (https://github.com/ONSdigital/dp-cli/tree/main/project_generation/content/templates/controller), as otel has been implemented in some of those already e.g. dp-frontend-homepage-controller

File: project_generation/content/templates/event/service/service.go.tmpl
Comment (HuwONS) [reply project_generation/content/templates/event/service/service.go.tmpl]: I've fixed this and added in the instrumentation here now, but am having running the project including in develop

**Review by franmoore05 — APPROVED** [review]

**Pull request merged**
